### PR TITLE
catboost: 0.24.4 (new formula)

### PR DESCRIPTION
**Pull request by cdalvaro**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/catboost.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+  livecheck do
+    url :stable
+    strategy :github_latest
+  end
+
+  def install
+    cd "#{buildpath}/catboost/app" do
+      ENV["YA_CACHE_DIR"] = "./.ya"
+      system "../../ya", "make", "-r", "-o", "#{buildpath}/build"
```
Comment (carlocab) [inline Formula/catboost.rb:16]: `ya` looks like it downloads something. I think that should be declared as a `resource` instead.

File: Formula/catboost.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      bin.install "#{buildpath}/build/catboost/app/catboost"
+    end
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/catboost --version")
```
Comment (carlocab) [inline Formula/catboost.rb:22]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

**Comment by carlocab** [comment]
Thanks for your PR, @cdalvaro.

File: Formula/catboost.rb (hunk 1 above)
Comment (carlocab) [reply Formula/catboost.rb:16]: Actually, it looks like it downloads a binary executable. We should build that from source instead.

Comment (cdalvaro) [reply Formula/catboost.rb:21]: OK!

I'll try to figure out how to do it!

Comment (cdalvaro) [reply Formula/catboost.rb:21]: It looks like `ya-bin` is not open source as explained here: https://github.com/catboost/catboost/issues/411#issuecomment-404250907

They provide makefiles to build catboost without `ya-bin`, but there is not makefile for macOS.

Can we continue building the formula as it is now?

We can create makefiles manually with `ya make --makefile` and add them as patches, but we need a makefile for x86_64 and Apple Silicon architectures.

Comment (SMillerDev) [reply Formula/catboost.rb:21]: If it pulls in closed source dependencies this is not a good fit for homebrew-core.

Please consider hosting it in a personal tap, which is very easy to do: https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap

Comment (cdalvaro) [reply Formula/catboost.rb:21]: OK!

No problem, I already has my own tap for these cases.

I'll open a new PR if the catboost team change the way they compile it source code.

**Comment by cdalvaro** [comment]
I close this PR as suggested by @SMillerDev in https://github.com/Homebrew/homebrew-core/pull/69930#discussion_r566601322 until there is an open source builder for catboost

If someone really needs this (as it is my case), the formula is available at my personal tap [cdalvaro/homebrew-tap](https://github.com/cdalvaro/homebrew-tap)

You can install it as:

```
brew install cdalvaro/tap/catboost
```

**Comment by carlocab** [comment]
Thanks again for your PR. Given that this is currently not suitable for homebrew/core, I'm closing this.

Feel free to open a new one when you are able to build this using only open-source software.

**Pull request closed without merging**